src/main/java/gregtech/api/metatileentity/multiblock/MultiblockControllerBase.java

```diff
@@ -317,6 +324,19 @@ public void addInformation(ItemStack stack, @Nullable World player, List<String>
         tooltip.add(I18n.format("gregtech.machine.multiblock.universal.controller_information", I18n.format(getMetaFullName())));
     }
 
+    @Override
+    public boolean onRightClick(EntityPlayer playerIn, EnumHand hand, EnumFacing facing, CuboidRayTraceResult hitResult) {
+        if (super.onRightClick(playerIn, hand, facing, hitResult))
+            return true;
+
+        if (!this.isStructureFormed() && playerIn.isSneaking() && this.rightClickDelayTimer == 0 && this.getWorld().isRemote) {
```

Comment:
This should check if the player's hand is empty. Otherwise, attempting to place blocks on top of the controller will cycle the preview.